boost/boost.bzl

```diff
@@ -136,14 +136,25 @@ def boost_deps():
             ],
         )
 
+    if "com_github_facebook_zstd" not in native.existing_rules():
+        http_archive(
+            name = "com_github_facebook_zstd",
+            urls = [
+                "https://github.com/facebook/zstd/releases/download/v1.4.4/zstd-1.4.4.tar.gz",
+            ],
+            sha256 = "59ef70ebb757ffe74a7b3fe9c305e2ba3350021a918d168a046c6300aeea9315",
+            build_file = "@com_github_nelhage_rules_boost//:BUILD.zstd",
+            strip_prefix = "zstd-1.4.4",
+        )
+
     if "boost" not in native.existing_rules():
         http_archive(
             name = "boost",
             build_file = "@com_github_nelhage_rules_boost//:BUILD.boost",
-            sha256 = "da3411ea45622579d419bfda66f45cd0f8c32a181d84adfa936f5688388995cf",
-            strip_prefix = "boost_1_68_0",
+            sha256 = "d73a8da01e8bf8c7eda40b4c84915071a8c8a0df4a6734537ddde4a8580524ee",
+            strip_prefix = "boost_1_71_0",
             urls = [
-                "https://%s.dl.sourceforge.net/project/boost/boost/1.68.0/boost_1_68_0.tar.gz" % m
-                for m in SOURCEFORGE_MIRRORS
+                "https://dl.bintray.com/boostorg/release/1.71.0/source/boost_1_71_0.tar.bz2",
             ],
+            patch_cmds = [ "rm -f doc/pdf/BUILD", ],
```

Comment:
This is quite gross, but `bazel` on HFS+ gets confused by the presence of `doc/pdf/build`. There definitely might be a better way to handle this. Open to suggestions...

On a case-sensitive filesystem this doesn't affect the filesystem status at all because there is no file named `doc/pdf/BUILD`.

I haven't tested on Windows and I don't know if they current implementation requires `bash` or not. If we currently work on windows and don't currently require `bash`, we should be able to add an empty `patch_cmds_win` so that we don't run anything.

Reply:
This seems as good a solution as any, to me. Case-insensitive filesystems are a pain. It might be worth filing an upstream issue asking Boost to avoid the conflict, though.